### PR TITLE
Ability to provide a CN format in PKI providers

### LINKED ISSUE #2116: pki: Add the ability to "format" the CN
When using a PKI provider, CN can be either the MAC address of the device or the PID used to log in.
In some cases, it might be useful to format the CN to either prefix or suffix something so that EAP configuration handling can be easier.

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description
Title says it all

# Impacts
Client generated certs for EAP-TLS

# Issue
fixes #2116 

# Delete branch after merge
YES

# NEWS file entries
## Bug Fixes
* pki: Add the ability to "format" the CN (#2116)

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Form/Config/PKI_Provider/packetfence_pki.pm [hunk 1]
```diff
@@ -138,6 +138,16 @@ has_field 'cn_attribute' => (
     },
 );
 
+has_field 'cn_format' => (
+    type    => 'Text',
+    label   => 'Common Name Format',
+    default => '%s',
+    tags    => {
+        after_element   => \&help,
+        help            => 'Defines how the common name will be formated. %s will expand the defined Common Name Attribute value',
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/PKI_Provider/packetfence_pki.pm]: Not 100% sure but I think its 'will expand to the defined'

File: html/pfappserver/lib/pfappserver/Form/Config/PKI_Provider/scep.pm [hunk 2]
```diff
@@ -120,9 +120,19 @@ has_field 'cn_attribute' =>
              help => 'Defines what attribute of the node to use as the common name during the certificate generation.' },
   );
 
+has_field 'cn_format' => (
+    type    => 'Text',
+    label   => 'Common Name Format',
+    default => '%s',
+    tags    => {
+        after_element   => \&help,
+        help            => 'Defines how the common name will be formated. %s will expand the defined Common Name Attribute value',
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/PKI_Provider/scep.pm]: same here

**Comment by dwlfrth** [comment]
Fixed buddy

**Comment by julsemaan** [comment]
merged buddy

**Pull request merged**